handlers/aws/handler.py

```diff
@@ -4,36 +4,114 @@
+        event_input = config.get_input_by_type_and_id("sqs", lambda_event["Records"][0]["eventSourceARN"])
+        if not event_input:
+            logger.error(f'no input set for {lambda_event["Records"][0]["eventSourceARN"]}')
+
+            return "not input set"
+
+        logger.info("input", extra={"type": event_input.type, "id": event_input.id})
+
+        sent_event: int = 0
+        composite_shipper: CompositeShipper = CompositeShipper()
 
-    if trigger_type == "sqs":
         try:
-            for es_event, offset in _handle_sqs_event(lambda_event):
-                _enrich_event(es_event, event_type, dataset, namespace)
-                shipper.send(es_event)
+            for output_type in event_input.get_output_types():
+                if output_type == "elasticsearch":
+                    logger.info("setting ElasticSearch shipper")
+                    output: ElasticSearchOutput = event_input.get_output_by_type("elasticsearch")
+                    shipper: CommonShipper = ShipperFactory.create_from_output(
+                        output_type="elasticsearch", output=output
+                    )
+                    composite_shipper.add_shipper(shipper=shipper)
+
+            for es_event, last_ending_offset, current_sqs_record, current_s3_record in _handle_sqs_event(
+                config, lambda_event
+            ):
+                logger.debug("es_event", extra={"es_event": es_event})
+
+                composite_shipper.send(es_event)
+                sent_event += 1
+
+                if (
+                    lambda_context is not None
+                    and lambda_context.get_remaining_time_in_millis() < _completion_grace_period
+                ):
+                    sqs_continuing_queue = os.environ["SQS_CONTINUE_URL"]
+
+                    logger.info(
+                        "lambda is going to shutdown, continuing on dedicated sqs queue",
+                        extra={"sqs_queue": sqs_continuing_queue, "sent_event": sent_event},
+                    )
+
+                    composite_shipper.flush()
+
+                    _handle_sqs_continuation(
+                        sqs_continuing_queue=sqs_continuing_queue,
+                        lambda_event=lambda_event,
+                        event_input_id=event_input.id,
+                        last_ending_offset=last_ending_offset,
+                        current_sqs_record=current_sqs_record,
+                        current_s3_record=current_s3_record,
+                        config_yaml=config_yaml,
+                    )
+
+                    return
+
+            composite_shipper.flush()
+            logger.info("lambda processed all the events", extra={"sent_event": sent_event})
 
         except Exception as e:
+            logger.exception("exception raised", exc_info=e)
+            logger.error("lambda exited with an error", extra={"sent_event": sent_event})
             return str(e)
```

Comment:
Nit. Instead of having so many try/except blocks intended to avoid the handler to raise exceptions, maybe we could have just a decorator that catches and logs all exceptions in the handler. We could still have smaller specific try blocks if needed for some specific case.